File: temp/LibrarySample/rudder-unity-client/Event/RudderEventBuilder.cs
﻿using System;
using com.rudderlabs.unity.library.Event.Property;

namespace com.rudderlabs.unity.library.Event
{
    public class RudderEventBuilder
    {
        private RudderProperty rudderProperty;
        public RudderEventBuilder SetRudderProperty(RudderProperty rudderProperty)
        {
            this.rudderProperty = rudderProperty;
            return this;
        }

        public RudderEventBuilder SetRudderProperty(RudderPropertyBuilder propertyBuilder)
        {
            this.rudderProperty = propertyBuilder.Build();
            return this;
        }

        private string eventName;
        public RudderEventBuilder SetEventName(string eventName)
        {
            this.eventName = eventName;
            return this;
        }

        private string userId;
        public RudderEventBuilder SetUseId(string userId)
        {
            this.userId = userId;
            return this;
        }

        private string channel;
        public RudderEventBuilder SetChannel(string channel)
        {
            this.channel = channel;
            return this;
        }

        public RudderEvent Build()
        {
            RudderEvent rudderEvent = new RudderEvent();
            rudderEvent.message.properties = rudderProperty;
            rudderEvent.message.eventName = eventName;
            rudderEvent.message.userId = userId;
            rudderEvent.message.channel = channel;
            return rudderEvent;
        }
    }
}

File: old-references/rudder-client-android-kotlin-old/src/main/java/com/rudderlabs/android/library/models/RudderEventBuilder.kt
package com.rudderlabs.android.library.models

import com.rudderlabs.android.library.models.porperties.RudderProperty
import com.rudderlabs.android.library.models.porperties.RudderPropertyBuilder

class RudderEventBuilder {
    private var rudderProperty: RudderProperty? = null
    fun setProperty(rudderProperty: RudderProperty): RudderEventBuilder {
        this.rudderProperty = rudderProperty
        return this
    }

    fun setPropertyMap(property: Map<String, Any?>): RudderEventBuilder {
        if (this.rudderProperty == null) this.rudderProperty = RudderProperty()
        this.rudderProperty!!.addProperties(property)
        return this
    }

    fun setPropertyBuilder(builderRudder: RudderPropertyBuilder): RudderEventBuilder {
        this.rudderProperty = builderRudder.build()
        return this
    }

    private var event: String? = null
    fun setEvent(event: String): RudderEventBuilder {
        this.event = event
        return this
    }

    private var userId: String? = null
    fun setUserId(userId: String): RudderEventBuilder {
        this.userId = userId
        return this
    }

    private var channel: String? = null
    fun setChannel(channel: String): RudderEventBuilder {
        this.channel = channel
        return this
    }

    fun build(): RudderEvent {
        return RudderEvent().initiate(RudderEventTemplate.getTemplate())
                .also {
                    it.message?.properties = this.rudderProperty
                    it.message?.userId = this.userId
                    it.message?.event = this.event
                    it.message?.channel = this.channel
                }
    }
}